### ISSUE TITLE
ximgproc: unstable test ximpgroc_Edgeboxes.regression

### ISSUE THREAD

**Issue by alalek** [opencv/opencv_contrib#2346]
On ~~master~~ both 3.4/master branches. Disabled.

relates #2178
https://github.com/opencv/opencv_contrib/pull/2178#issuecomment-541354674

Tolerance relaxing from 1e-8 doesn't help.

[1](http://pullrequest.opencv.org/buildbot/builders/precommit-contrib_linux64/builds/13291/steps/test_ximgproc/logs/stdio):
```
 RUN      ] ximpgroc_Edgeboxes.regression
/build/precommit-contrib_linux64/opencv_contrib/modules/ximgproc/test/test_edgeboxes.cpp:41: Failure
The difference between scores[0] and expectedScore is 0.048548191785812378, which exceeds 1e-8, where
scores[0] evaluates to 0.43887743353843689,
expectedScore evaluates to 0.48742562532424927, and
1e-8 evaluates to 1e-08.
[  FAILED  ] ximpgroc_Edgeboxes.regression (3005 ms)
```

[2](http://pullrequest.opencv.org/buildbot/builders/precommit-contrib_linux64/builds/13292/steps/test_ximgproc/logs/stdio):
```
[ RUN      ] ximpgroc_Edgeboxes.regression
/build/precommit-contrib_linux64/opencv_contrib/modules/ximgproc/test/test_edgeboxes.cpp:41: Failure
The difference between scores[0] and expectedScore is 0.1133061945438385, which exceeds 0.05f, where
scores[0] evaluates to 0.37411943078041077,
expectedScore evaluates to 0.48742562532424927, and
0.05f evaluates to 0.05000000074505806.
[  FAILED  ] ximpgroc_Edgeboxes.regression (2008 ms)
```

[3](http://pullrequest.opencv.org/buildbot/builders/precommit-contrib_linux64_no_opt/builds/10251/steps/test_ximgproc/logs/stdio):
```
[ RUN      ] ximpgroc_Edgeboxes.regression
/build/precommit-contrib_linux64_no_opt/opencv_contrib/modules/ximgproc/test/test_edgeboxes.cpp:41: Failure
The difference between scores[0] and expectedScore is 0.015961050987243652, which exceeds 1e-8, where
scores[0] evaluates to 0.47146457433700562,
expectedScore evaluates to 0.48742562532424927, and
1e-8 evaluates to 1e-08.
[  FAILED  ] ximpgroc_Edgeboxes.regression (6557 ms)
```

**Comment by alalek** [comment]
3.4 branch is affected too:

- [1](http://pullrequest.opencv.org/buildbot/builders/3_4-contrib-win64-vc14/builds/528)

**Comment by saskatchewancatch** [comment]
That's unfortunate. It was definitely stable on my machine.
@alalek looks like you've found out what's causing instability on some environments and have merged a fix in.
Perhaps I will wait for that to get merged from 3.4 into master, and then adjust the Test case where necessary??

Thank you for addressing that in the Edgebox implementation.🙏
